filters/ema.hpp

```diff
@@ -1,109 +1,156 @@
-#ifndef SERENITY_EXPONENTIAL_MOVING_AVERAGE_HPP
-#define SERENITY_EXPONENTIAL_MOVING_AVERAGE_HPP
+#ifndef SERENITY_EXPONENTIAL_MOVING_AVERAGE_FILTER_HPP
+#define SERENITY_EXPONENTIAL_MOVING_AVERAGE_FILTER_HPP
 
 #include <math.h>
 
+#include <bits/unique_ptr.h>
+
+#include <glog/logging.h>
+
+
 #include "messages/serenity.hpp"
 
+#include "serenity/executor_map.hpp"
+#include "serenity/executor_set.hpp"
 #include "serenity/serenity.hpp"
 
-#include "stout/option.hpp"
+#include "stout/lambda.hpp"
+#include "stout/nothing.hpp"
 
 namespace mesos {
 namespace serenity {
 
-#define DEFAULT_EMA_FILTER_ALPHA 0.2
-
 /**
- * EMAFilter returns Exponential Moving Average (double) for
- * given input.
+ * The smaller alpha becomes, the longer your moving average is.
+ * It becomer smoother, but less reactive to new samples.
  */
-template <typename In>
-class EMAFilter : public Consumer<In>, public Producer<double> {
- public:
-  explicit EMAFilter(Option<double> _alpha) {
-    if (_alpha.isSome()) {
-      setAlpha(_alpha.get());
-    } else {
-      setAlpha(DEFAULT_EMA_FILTER_ALPHA);
-    }
-  }
+constexpr double_t DEFAULT_EMA_FILTER_ALPHA = 0.2;
```

Comment:
We need it in several places, so it's better to leave it in filters/ema.hpp domain.